### ISSUE TITLE
PID Tuning

### ISSUE THREAD

**Issue by haudamekki** [makerbase-mks/MKS-SERVO42B#12]
How could I tune the PID values? Is there a serial interface like the MisfitTech Smart Stepper has?

**Comment by lonelymyp** [comment]
@haudamekki 
 No. This is a bad copy of the MisfitTech old version, there is a lack of many necessary things.

**Comment by dzid26** [comment]
> @haudamekki
> No. This is a bad copy of the MisfitTech old version, there is a lack of many necessary things.

What is it lacking apart from serial?
I think more similar copy is atmel based Servo42A https://github.com/makerbase-mks/MKS-SERVO42A

**Comment by lonelymyp** [comment]
> What is it lacking apart from serial?
@dzid26 
For example, the code for control on uart, [commands.cpp](https://github.com/Misfittech/nano_stepper/blob/master/firmware/stepper_nano_zero/commands.cpp) cut out completely.

The code for different modes of PID operation was ignored. 
There are 3 different modes in the original, SIMPLE PID, VELOCITY PID, POSITIONAL PID.
Only 1 mode was left here, SIMPLE PID.

Cut-out holding current control. The holding current cannot be set in the menu, it is rigidly tied to the operating current.
In this case, in fact, the holding current is set, and the operating current is simply 2 times higher.
https://github.com/makerbase-mks/MKS-SERVO42B/issues/7
